src/App.js

```diff
@@ -1,11 +1,52 @@
-import React from 'react';
+import React, {Component} from 'react';
+import {BrowserRouter as Router, Route, Switch, Redirect} from "react-router-dom";
+import './App.css';
+import {Login} from "../src/components/Login";
+import {Register} from "./components/Register";
+import {Welcome} from "./components/Welcome";
+import {Page404} from "./components/Page404";
+import {MainPage} from "./components/MainPage";
 
-function App() {
+const withAuth = (Component, auth) => {
+    return (props) => {
+        return <Component {...props} auth={auth}/>
+    };
+};
+
+class App extends React.Component {
+    constructor(props) {
+        super(props);
+
+        const tokens = localStorage.getItem('AUTH');
+
+        this.state = {
+            tokens: tokens || null
+        };
+    }
+
+    auth = (tokens) => {
+        this.setState({tokens: tokens})
+    }
+
+  render() {
+    const {tokens} = this.state;
     return (
-        <div>
-            This homework requires a Hero!
+        <div className={"App"}>
+              <Router>
+                 {!tokens &&<Switch>
+                      <Route exact={true} path="/" component={Welcome}/>
+                      <Route path="/register" component={withAuth(Register, this.auth)}/>
+                      <Route path="/login" component={withAuth(Login, this.auth)}/>
+                      <Redirect to="/"/>
+                      <Route component={Page404}/>
```

Comment:
I think `Router` after `Redirect` is unreachable, but maybe I don't understand smth